RevenueCatUI/Modifiers/ViewExtensions.swift

```diff
@@ -272,14 +288,29 @@ private struct RoundedCorner: Shape {
 
 // MARK: - Preference Keys
 
-/// `PreferenceKey` for keeping track of a view dimension.
-private struct ViewDimensionPreferenceKey: PreferenceKey {
+@available(iOS 13.0, tvOS 13.0, macOS 10.15, watchOS 6.2, *)
+private protocol ViewDimensionPreferenceKey: PreferenceKey where Value == CGFloat {}
 
-    typealias Value = CGFloat
+/// `PreferenceKey` for keeping track of a view width.
+private struct ViewWidthPreferenceKey: ViewDimensionPreferenceKey {
 
     static var defaultValue: Value = 10
 
-    static func reduce(value: inout Value, nextValue: () -> Value) {
+    static func reduce(value: inout CGFloat, nextValue: () -> CGFloat) {
```

Comment:
Is it possible to move this static method to the protocol considering it's repeated between width and height?

Reply:
Done!